src/formatDate.js

```diff
@@ -8,7 +8,62 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const sep = fromFormat[fromFormat.length - 1];
+  const parts = date.split(sep);
+  const map = {};
+  const sepTo = toFormat[toFormat.length - 1];
+  const resultParts = [];
+
+  if (parts.length !== 3) {
+    return date;
+  }
+
+  function pad2(x) {
+    const s = String(x);
+
+    return s.length === 1 ? '0' + s : s;
+  }
```

Comment:
```suggestion
 String(x).padStart(2, '0');
```
don't reinvent a wheel :)